code/modules/recycling/disposal-construction.dm

```diff
@@ -170,10 +170,13 @@
 			nicetype = "pipe"
 
 	var/turf/T = loc
-	if(T.intact)
+	if(T.intact && istype(T, /turf/simulated/floor))

```

Comment:
There are places where disposals run through space and I didn't want to block replacing/building there.